daemon/ucrednix.go

```diff
@@ -0,0 +1,68 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"fmt"
+	"net"
+	sys "syscall"
+)
+
+type ucrednixAddr struct {
+	*net.UnixAddr
+	ucred *sys.Ucred
+}
+
+func (wa *ucrednixAddr) String() string {
+	return fmt.Sprintf("%d:%s", wa.ucred.Uid, wa.UnixAddr.String())
+}
```

Comment:
yeh, that's the kind of thing i'd need to check; for example, does RemoteAddr do a dns lookup? if so there's a way for somebody to make a reverse lookup inject something that looks like a uid in there, as things stand right now. Etc.

I guess I could avoid this whole thing by stomping on RemoteAddr for every connection, not just unix-ish ones.